src/transaction/log_page_async_fetcher.cpp

```diff
@@ -49,11 +49,15 @@ namespace cublog
 
   async_page_fetcher::async_page_fetcher ()
   {
-    cubthread::manager *thread_manager = cubthread::get_manager ();
+    cubthread::manager *const thread_manager = cubthread::get_manager ();
 
     const auto thread_count = std::thread::hardware_concurrency ();
-    m_threads = thread_manager->create_worker_pool (thread_count, thread_count, "async_page_fetcher_worker_pool",
-		nullptr, thread_count, false /*debug_logging*/);
+    m_worker_pool_context_manager.reset (
+	    new cubthread::system_worker_entry_manager (TT_WORKER));
```

Comment:
Although the name may not be fortunate, TT_WORKER is the type of threads that execute USER operations. For system operations, the default type would be TT_DAEMON.

It is probably better to add a new thread type: TT_SYSTEM_WORKER.

Reply:
Added.
I pasted in the explanations above as comments.
But the semantics & separation is not really clear and could be better.

Reply:
True, it could, but I think your comments are enough for now :).